app/models/concerns/audit_associations.rb

```diff
@@ -0,0 +1,42 @@
+module AuditAssociations
+  def changes_to_save
+    unless audited_options[:associations].blank?
+      super.merge(associated_changes)
+    else
+      super
+    end
+  end
+
+  private
+
+  def associated_changes
+    associations = Array.wrap(audited_options[:associations])
+    associations.inject({}) do |changes_hash, association_name|
+      association_name = association_name.to_s.singularize
+      association_ids = "#{association_name}_ids"
+
+      if send("#{association_ids}_changed?")
+        association_class = association_name.capitalize.constantize
+        change = send("#{association_ids}_change")
+        change_ids = change.flatten.uniq
+
+        human_readable_attr = human_readable_attr(association_class)
+        id_name_map = association_class.where(id: change_ids).select(:id, human_readable_attr).inject({}) { |r,p| r.merge(p.id => p.send(human_readable_attr)) }
```

Comment:
I think it's better to load all attributes and call `.to_label` that renders the string representation of the object, we can't tell what all attributes are required for this representation to be rendered